spec version check fails

spec version check currently fails on CI. I think I've seen this when using an older pip: `pip install -e .` thinks the project uses setuptools (either because it does not support pyproject.toml or because it gets confused by our decoy setup.py).

This could be fixed by
* fixing #2070 or
* updating pip in the workflow

I assume the first one is enough, but it's possible that both are needed (if pip is old enough then it literally can't install pyproject.toml projects with "-e"). I suggest trying #2070 first.

```
python3 -m pip install -e .
Obtaining file:///home/runner/work/python-tuf/python-tuf
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'done'
Collecting securesystemslib>=0.22.0
  Downloading securesystemslib-0.23.0-py3-none-any.whl (107 kB)
Requirement already satisfied: requests>=2.1[9](https://github.com/theupdateframework/python-tuf/runs/8057985831?check_suite_focus=true#step:4:10).1 in /usr/lib/python3/dist-packages (from tuf==2.0.0) (2.22.0)
Installing collected packages: securesystemslib, tuf
  Running setup.py develop for tuf
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/home/runner/work/python-tuf/python-tuf/setup.py'"'"'; __file__='"'"'/home/runner/work/python-tuf/python-tuf/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps --user --prefix=
         cwd: /home/runner/work/python-tuf/python-tuf/
    Complete output (3 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
    ModuleNotFoundError: No module named 'setuptools'
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/home/runner/work/python-tuf/python-tuf/setup.py'"'"'; __file__='"'"'/home/runner/work/python-tuf/python-tuf/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps --user --prefix= Check the logs for full command output.
Error: Process completed with exit code 1.
```

--- a/setup.py
+++ b/setup.py
@@ -1,4 +1,0 @@
-# This file exists to keep dependabot happy:
-# https://github.com/dependabot/dependabot-core/issues/4483
-from setuptools import setup
-setup()

